### ISSUE TITLE
Fuzzy search not finding Juncus bulbosos

### ISSUE THREAD

**Issue by mdoering** [CatalogueOfLife/backend#992]
even though _Juncus bulbosus_ exists:

https://api.catalogueoflife.org/dataset/3LR/nameusage/search?q=Juncus%20bulbosus
https://api.catalogueoflife.org/dataset/3LR/nameusage/search?fuzzy=true&q=Juncus%20bulbosos

**Comment by mdoering** [comment]
The fuzzy search is apparently implemented to only rely on the scientficName normalizer and is no true generic fuzzy search.
@ayco-at-naturalis is that correct? How/where would we best add a true generic 1 char distance fuzzy search, into the FuzzyMatcher and its subclasses?

**Comment by mdoering** [comment]
Other reported examples: Lilium rhodopeum <-> Lilium rhodopaeum

**Comment by ayco-at-naturalis** [comment]
Yes, that is correct. I did propose to **also** add a true Elasticsearch-native fuzzy search functionality (or maybe combine it with the normalizer-based fuzzy search), but at the time it was deemed not necessary. I could ask my manager to give me some time to build it in after all, if it's desirable, but it would be next year I think.

**Comment by ayco-at-naturalis** [comment]
If you want to do it yourself, it's been some time, but I guess I would create a FuzzyQuery class in the es.query package that more or less models the ES query DSL:

```
{
  "query": {
    "fuzzy": {
      "user.id": {
        "value": "ki",
        "fuzziness": "AUTO",
        "max_expansions": 50,
        "prefix_length": 0,
        "transpositions": true,
        "rewrite": "constant_score"
      }
    }
  }
}
```

And then add that query to the DisMaxQueries in the (subclasses of) QMatcher in a way that feels reasonable.

See https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-fuzzy-query.html

**Comment by mdoering** [comment]
Thanks, I'll give it a shot in the next weeks
